core/src/transform/free_vars.rs

```diff
@@ -60,26 +60,36 @@ impl CollectFreeVars for RichTerm {
 
                 free_vars.extend(fresh);
             }
-            Term::Let(id, t1, t2, attrs) => {
+            Term::Let(bindings, body, attrs) => {
                 let mut fresh = HashSet::new();
 
-                if attrs.rec {
-                    t1.collect_free_vars(&mut fresh);
-                } else {
-                    t1.collect_free_vars(free_vars);
+                // FIXME: is this correct? `let x = x in y` should have x as a free var, no?
```

Comment:
Yes, and yes. Note that nothing is ever removed from `free_vars`, just from `fresh` before extending `free_vars`.So if the let isn't recursive, we will collect the free variables of `x` in `free_vars` - so, this is the set `{x}` - and it will stay that way. Put differently: `fresh` corresponds to the scope where the bound variable is available (so it's not a free variable of the overall let-binding). `free_vars` is the set of free variables of the whole let-binding.